.github/workflows/ci.yaml

```diff
@@ -8,40 +8,39 @@ on:
     branches:
       - master
 
+env:
+  POETRY_VIRTUALENVS_CREATE: "false"
+
 jobs:
   default:
-    name: "${{ matrix.step.env }} '${{ matrix.step.command }}' @ py${{ matrix.step.python }})"
+    name: "${{ matrix.step.env }} '${{ matrix.step.command }}' (@py${{ matrix.step.python }})"
     runs-on: ubuntu-latest
     strategy:
       matrix:
         step: [
           # original from drone CI startlark file
           # {env: "pytest", python: "3.6", extras: "main,dev", command: "pytest tests/"},  # disable or drop support?
-          {env: "pytest", python: "3.7", extras: "main,dev", command: "pytest tests/"},
-          {env: "pytest", python: "3.8", extras: "main,dev", command: "pytest tests/"},
-          {env: "pytest", python: "3.9", extras: "main,dev", command: "pytest tests/"},
-          {env: "main", python: "3.8", extras: "main,dev", command: "flakeheaven lint"},
-          {env: "typing", python: "3.8", extras: "main,dev", command: "mypy --ignore-missing-imports --allow-redefinition flakeheaven/ > /dev/null || yes | mypy --install-types && mypy --ignore-missing-imports --allow-redefinition flakeheaven/"},
+          {env: "pytest", python: "3.7", extras: "", command: "pytest tests/"},
+          {env: "pytest", python: "3.8", extras: "", command: "pytest tests/"},
+          {env: "pytest", python: "3.9", extras: "", command: "pytest tests/"},
+          {env: "main", python: "3.8", extras: "", command: "flakeheaven lint"},
+          {env: "typing", python: "3.8", extras: "", command: "mypy --ignore-missing-imports --allow-redefinition flakeheaven/ > /dev/null || yes | mypy --install-types && mypy --ignore-missing-imports --allow-redefinition flakeheaven/"},
           # additional
-          {env: "docs", python: "3.8", extras: "docs", command: "sphinx-build -W docs docs/build"},
-          {env: "isort", python: "3.8", extras: "main,dev", command: "isort --check flakeheaven"},
+          {env: "docs", python: "3.8", extras: "-E docs", command: "sphinx-build -W docs docs/build"},
+          {env: "isort", python: "3.8", extras: "", command: "isort --check flakeheaven"},
         ]
     container:
-      image: python:${{ matrix.step.python }}-alpine
+      image: python:${{ matrix.step.python }}-bullseye
       env:
         COVERAGE_FILE: ".coverage.${{ matrix.step.env }}.${{ matrix.step.python }}"
     steps:
       - name: Check out repository
         uses: actions/checkout@v2
-      - name: Setup env
-        run: |
-          apk add \
-            gcc \
-            libc-dev
+      - name: Install poetry
+        run: pip install poetry
```

Comment:
I do not think installing poetry in the same env as the application is a good practice:we could be missing deps if they're installed for poetry and not flake heaven and we wouldn't notice

What is the advantage of using poetry here vs plain pip install? We're not using poetry build/publish...
The build system in project.toml allows pip to install flakeheaven



I'd either install poetry somewhere else or just use pip

Reply:
Hi, I have a working poetry workflow here: https://github.com/uSpike/rpcx/blob/main/.github/workflows/main.yml

It uses caching to not repeatedly install poetry/deps if not required, and also runs pre-commit hooks.  Feel free to copy.

Reply:
> I do not think installing poetry in the same env as the application is a good practice:we could be missing deps if they're installed for poetry and not flake heaven and we wouldn't notice
> 
> What is the advantage of using poetry here vs plain pip install? We're not using poetry build/publish... The build system in project.toml allows pip to install flakeheaven
> 
> I'd either install poetry somewhere else or just use pip

good point. I have enabled `poetry` venv so that flakeheaven is installed and run inside a venv (separate from poetry)

Reply:
> Hi, I have a working poetry workflow here: https://github.com/uSpike/rpcx/blob/main/.github/workflows/main.yml
> 
> It uses caching to not repeatedly install poetry/deps if not required, and also runs pre-commit hooks. Feel free to copy.

thanks for the example! it looks really good, to keep the scope of this PR smaller I would prefer to add it in a follow up PR, we did not have caching previously anyways, so adding caching seems like a new feature that can be added separately